internal/repository/hook.go

```diff
@@ -0,0 +1,29 @@
+// SPDX-License-Identifier: Apache-2.0
+
+package repository
+
+import (
+	"fmt"
+	"os"
+	"path"
+)
+
+type HookType string
+
+var HookPrePush = HookType("pre-push")
+
+func (r *Repository) UpdateHook(hookType HookType, content []byte) error {
+	conf, err := r.r.Config()
+	if err != nil {
+		return fmt.Errorf("reading repository config: %w", err)
+	}
+
+	repoRoot := conf.Core.Worktree
+	hookFile := path.Join(repoRoot, ".git", "hooks", string(hookType))
+
+	if err := os.WriteFile(hookFile, content, 0o700); err != nil {
```

Comment:
We should at the very least abort if a pre-push hook already exists. We don't want to nuke other hooks the user may have installed.

Reply:
As a side note, hooks may not always be scripts, they can be any executable binary. So appending to existing hooks may not work as well.

Reply:
good point! i think we check and if so, we present the parts of the hook they ought to add to the script themself. Maybe a link to docs someplace.

Reply:
Good point. I added the necessary checks to not overwrite existing hooks. 
Do we want to provide a `--force` flag for `gittuf add-hooks` or are we good with the basic functionality for now?

Reply:
`--force` is fine to add I think, with a warning printed to stderr.